### PR TITLE
data display without login

### PR DESCRIPTION
**Pull request by RoseRompkxm**
support frontend to display the basic system info when no wallets are connected, which can improve user experience.

### CODE REVIEW

File: src/utilities/web3.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ import getNodeUrl from './getRpcUrl';
 
 const RPC_URL = getNodeUrl();
 
+// @todo: we will get the "No 'Access-Control-Allow-Origin' header is present on the requested resource"
+// error on the frontend in testnet when the wallet is unconnected, because the bsc official testnet endpoints
+// don't support CORS request right now, and we didn't find any working testnet endpoints for
+// HTTPProvider or WebSocketProvider neither.
```
Comment (kkirka) [inline src/utilities/web3.js:9]: Do mainnet endpoints support cors? Can we set up a cors proxy for testnet rpc?

Comment (RoseRompkxm) [reply src/utilities/web3.js:9]: ofc we can, but it needs some work in the backend, maybe we can start a new issue after, currently, this issue won't bother our users imo

File: src/components/Basic/BorrowTabs/RepayBorrowTab.js [hunk 2]
```diff
@@ -311,7 +311,7 @@ function RepayBorrowTab({ asset, settings, changeTab, onCancel, setSetting }) {
               )}
             </div>
             <Progress
-              percent={newBorrowPercent.toString(10)}
+              percent={newBorrowPercent.toNumber(10)}
```
Comment (kkirka) [inline src/components/Basic/BorrowTabs/RepayBorrowTab.js]:
```suggestion
              percent={newBorrowPercent.toNumber()}
```

File: src/components/Basic/ConnectButton.js [hunk 3]
```diff
@@ -304,18 +303,16 @@ function ConnectButton() {
...
-                      <span>Wallet Connect</span>
-                    </div>
-                    <img src={arrowRightImg} alt="arrow" />
+                <div
+                  className="flex align-center just-between metamask-connect-btn"
+                  onClick={() => onConnect(ConnectorNames.WalletConnect)}
+                >
+                  <div className="flex align-center">
+                    <img src={walletconnectImg} alt="coinbase wallet" />
+                    <span>Wallet Connect</span>
```
Comment (kkirka) [inline src/components/Basic/ConnectButton.js]:
```suggestion
                    <span>WalletConnect</span>
```

File: src/components/Dashboard/Market/SupplyMarket.js [hunk 4]
```diff
@@ -60,14 +60,14 @@ function SupplyMarket({ settings, suppliedAssets, remainAssets }) {
         toast.error({
           title: `Collateral Required`,
           description:
-            'You need to set collateral at least one asset for your borrowed assets. Please repay all borrowed asset or set other asset as collateral.'
+            'Please repay all borrowed asset or set other assets as collateral.'
```
Comment (kkirka) [inline src/components/Dashboard/Market/SupplyMarket.js]:
```suggestion
            'Please repay all borrowed assets or set other assets as collateral.'
```

File: src/components/Dashboard/Market/SupplyMarket.js [hunk 5]
```diff
@@ -60,14 +60,14 @@ function SupplyMarket({ settings, suppliedAssets, remainAssets }) {
...
         });
       }
     } else {
       toast.error({
         title: `Collateral Required`,
         description:
-          'You need to set collateral at least one asset for your borrowed assets. Please repay all borrowed asset or set other asset as collateral.'
+          'Please repay all borrowed asset or set other assets as collateral.'
```
Comment (kkirka) [inline src/components/Dashboard/Market/SupplyMarket.js]:
```suggestion
          'Please repay all borrowed assets or set other assets as collateral.'
```

**Review by kkirka — APPROVED** [review]

**Pull request merged**